src/server/scripts/Spells/spell_item.cpp

```diff
@@ -4092,6 +4092,62 @@ class spell_item_skyguard_blasting_charges : public SpellScript
     }
 };
 
+// 23595 - Luffa
+class spell_item_luffa : public SpellScript
+{
+    PrepareSpellScript(spell_item_luffa);
+    std::list<uint32> removeList;
+    std::list<ObjectGuid> casterList;
+
+    SpellCastResult CheckCast()
+    {
+        if (Unit* caster = GetCaster())
+        {
+            Unit::AuraApplicationMap const& auras = caster->GetAppliedAuras();
+            for (Unit::AuraApplicationMap::const_iterator itr = auras.begin(); itr != auras.end(); ++itr)
+            {
+                Aura const* aura = itr->second->GetBase();
+                if (aura->GetSpellInfo()->GetEffectMechanic(EFFECT_0) != MECHANIC_BLEED || aura->GetCasterLevel() > 60 || aura->GetSpellInfo()->IsPositive())
```

Comment:
`GetEffectMechanic(EFFECT_0)` only checks for `EFFECT_0`. You can check for all effects with `aura->GetSpellInfo()->GetAllEffectsMechanicMask() & (1 << MECHANIC_BLEED)`

I don't think saving to lists here is needed. `CheckCast()` is [used to override the usual result](https://github.com/avarishd/azerothcore-wotlk/blob/5a13809e8e9f491da95e13e47bab29bc718dd68d/src/server/game/Spells/Spell.cpp#L6072). returning `SPELL_FAILED_NOTHING_TO_DISPEL` here if we find a bleed by a `GetCasterLevel() > 60` is enough. The rest can be done in `HandleEffect`